### PR TITLE
Feat/dynamo db support

### PR DESCRIPTION
**Pull request by manojown**
#### What does this PR do?
 DynamoDb Support - #219

#### Env Required
There is few ways we can connect dynamodb with our app

1. AWS credentials config - if this setup is already done then wont need to specify any key in env,  require to pass `REGION`
2. Manually Pass the `AWS_ACCESS_KEY` , `AWS_SECRET_KEY` and `REGION` in env file.
3. Local setup - In Local setup needed to pass `DATABASE_URL` and `REGION` which suppose to be you `http://localhost:port` ( default port : 8000 ).

#### If this PR affects any API reference documentation, please share the updated endpoint references
Note - In other DB's indices prevent the duplication in table but in dynamodb it's not the case so have to take care of duplication by our self.

#### Local testing tools can help
`https://github.com/aaronshaf/dynamodb-admin` -  helps in investigate the data, meta and manual crud operation.

### CODE REVIEW

**Comment by lakhansamani** [comment]
@manojown there are some merge conflicts that need to be resolved.

**Comment by manojown** [comment]
here is one small-check, 

> @manojown there are some merge conflicts that need to be resolved.

Done @lakhansamani

**Review by lakhansamani — CHANGES_REQUESTED** [review]
Requested some changes
Plus we need to add env vars to our persistent store, by adding them here
https://github.com/authorizerdev/authorizer/blob/main/server/env/env.go

File: server/memorystore/required_env_store.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ import (
 // RequiredEnv holds information about required envs
 type RequiredEnv struct {
 	EnvPath            string `json:"ENV_PATH"`
+	REGION             string `json:"REGION"`
```
[the remark this answers is not in the document]
Comment (manojown) [reply server/memorystore/required_env_store.go:19]: sure, as per the discussion going to remove this changes as it's specific to dynamo only.

**Comment by manojown** [comment]
> Requested some changes Plus we need to add env vars to our persistent store, by adding them here https://github.com/authorizerdev/authorizer/blob/main/server/env/env.go
 
 @lakhansamani AWS_SECRET and AWS_ACCESS keys should be confidential, is this good idea to keep in persistent store? Region we can definitely add in persistent.  
 
WYT ?

**Comment by lakhansamani** [comment]
We also store other secrets in encrypted format so nothing wrong it. Only one change we need to make is encrypt redis store data and I am working on it

**Comment by manojown** [comment]
@lakhansamani what if user not specify any key's and wanted to load credentials from aws config ? 
should we force user to add keys in env ?

**Comment by lakhansamani** [comment]
Got it sure, we can allow user to load from aws config

**Comment by lakhansamani** [comment]
@manojown is this ready for review?
In that case you move from draft -> ready for review

**Comment by manojown** [comment]
@lakhansamani yes.

**Comment by manojMeesho** [comment]
> Requested some changes Plus we need to add env vars to our persistent store, by adding them here https://github.com/authorizerdev/authorizer/blob/main/server/env/env.go

@lakhansamani This comment is resolved right but it still appeared as pending.

**Comment by manojown** [comment]
> Requested some changes Plus we need to add env vars to our persistent store, by adding them here https://github.com/authorizerdev/authorizer/blob/main/server/env/env.go

@lakhansamani this comment is resolve right, but it still appeared as pending.

**Comment by lakhansamani** [comment]
Yes, it's resolved now.
Merging this on development and will be doing beta release soon.

**Comment by lakhansamani** [comment]
@manojown thank you for the contribution.
feel free to take fork from development branch for your new db integration.

**Pull request merged**